src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -565,5 +544,95 @@ internal void TakeSnapshot(bool outputDelta)
         {
             throw new NotSupportedException($"{methodName} is not supported for this metric type.");
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void HistogramUpdate(double number, bool hasBuckets, bool hasMinMax)
+        {
+            int i = hasBuckets
+                ? this.histogramBuckets.FindBucketIndex(number)
+                : 0;
+
+            var sw = default(SpinWait);
+            while (true)
+            {
+                if (Interlocked.Exchange(ref this.histogramBuckets.IsCriticalSectionOccupied, 1) == 0)
+                {
+                    // Lock acquired
+                    unchecked
+                    {
+                        this.runningValue.AsLong++;
+                        this.histogramBuckets.RunningSum += number;
+                        if (hasBuckets)
+                        {
+                            this.histogramBuckets.RunningBucketCounts[i]++;
+                        }
+
+                        if (hasMinMax)
+                        {
+                            this.histogramBuckets.RunningMin = Math.Min(this.histogramBuckets.RunningMin, number);
+                            this.histogramBuckets.RunningMax = Math.Max(this.histogramBuckets.RunningMax, number);
+                        }
+                    }
+
+                    // Release lock
+                    Interlocked.Exchange(ref this.histogramBuckets.IsCriticalSectionOccupied, 0);
+                    break;
+                }
+
+                sw.SpinOnce();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void HistogramSnapshot(bool outputDelta, bool hasBuckets, bool hasMinMax)
+        {
+            var sw = default(SpinWait);
+            while (true)
+            {
+                if (Interlocked.Exchange(ref this.histogramBuckets.IsCriticalSectionOccupied, 1) == 0)
+                {
+                    // Lock acquired
+                    this.snapshotValue.AsLong = this.runningValue.AsLong;
+                    this.histogramBuckets.SnapshotSum = this.histogramBuckets.RunningSum;
+                    if (hasMinMax)
+                    {
+                        this.histogramBuckets.SnapshotMin = this.histogramBuckets.RunningMin;
+                        this.histogramBuckets.SnapshotMax = this.histogramBuckets.RunningMax;
+                    }
+
+                    if (outputDelta)
+                    {
+                        this.runningValue.AsLong = 0;
+                        this.histogramBuckets.RunningSum = 0;
+
+                        if (hasMinMax)
+                        {
+                            this.histogramBuckets.RunningMin = 0;
```

Comment:
In case of delta temporality, `RunninMin` and `RunningMax` should be set `double.PositiveInfinity` and `double.NegativeInfinity` respectively.